payments-core/src/main/java/com/stripe/android/payments/bankaccount/ui/CollectBankAccountViewModel.kt

```diff
@@ -107,55 +123,63 @@ internal class CollectBankAccountViewModel @Inject constructor(
         }
     }
 
-    private suspend fun finishWithResult(result: CollectBankAccountResult) {
+    private suspend fun finishWithResult(result: CollectBankAccountResultInternal) {
         _viewEffect.emit(CollectBankAccountViewEffect.FinishWithResult(result))
     }
 
-    private fun finishWithFinancialConnectionsSession(financialConnectionsSession: FinancialConnectionsSession) {
+    private fun finishWithFinancialConnectionsSession(
+        financialConnectionsSession: FinancialConnectionsSession
+    ) {
         viewModelScope.launch {
-            retrieveStripeIntent(
-                args.publishableKey,
-                args.clientSecret
-            ).onSuccess { stripeIntent ->
-                finishWithResult(
-                    Completed(
-                        CollectBankAccountResponse(
-                            intent = stripeIntent,
-                            financialConnectionsSession = financialConnectionsSession
-                        )
+            finishWithResult(
+                CollectBankAccountResultInternal.Completed(
+                    CollectBankAccountResponseInternal(
+                        intent = args.clientSecret?.let { clientSecret ->
+                            retrieveStripeIntent(
+                                args.publishableKey,
+                                clientSecret
+                            ).onFailure {
+                                finishWithError(it)
+                            }.getOrNull()
+                        },
+                        financialConnectionsSession = financialConnectionsSession
```

Comment:
It might be difficult to understand errors vs deferred intents here. Also we don't want to run `finishWithError` inside `finishWithResult`. Something like this migth be better:

```
private fun finishWithFinancialConnectionsSession(
        financialConnectionsSession: FinancialConnectionsSession
    ) {
        viewModelScope.launch {
            val retrieveIntentResult: Result<StripeIntent?> =
                when (val intentSecret = args.clientSecret) {
                    // client secret is null for deferred intents.
                    null -> Result.success(null)
                    else -> retrieveStripeIntent(args.publishableKey, intentSecret)
                }
            retrieveIntentResult
                .onFailure { finishWithError(it) }
                .onSuccess { intent -> 
                    finishWithResult(
                        CollectBankAccountResultInternal.Completed(
                            CollectBankAccountResponseInternal(
                                intent,
                                financialConnectionsSession = financialConnectionsSession
                            )
                        )
                    )
                }
        }
    }
```